test/e2e-go/upgrades/send_receive_upgrade_test.go

```diff
@@ -141,23 +141,41 @@ func testAccountsCanSendMoneyAcrossUpgrade(t *testing.T, templatePath string) {
 	var pingTxids []string
 	var pongTxids []string
 
+	pongWalletHandle, err := pongClient.GetUnencryptedWalletHandle()
+	a.NoError(err)
+	pingWalletHandle, err := pingClient.GetUnencryptedWalletHandle()
+	a.NoError(err)
 	startTime := time.Now()
+	var lastTxnSendRound uint64
 	for curStatus.LastVersion == initialStatus.LastVersion {
-		pongTx, err := pongClient.SendPaymentFromUnencryptedWallet(pongAccount, pingAccount, transactionFee, amountPongSendsPing, GenerateRandomBytes(8))
-		a.NoError(err, "fixture should be able to send money (pong -> ping)")
-		pongTxids = append(pongTxids, pongTx.ID().String())
+		iterationStartTime := time.Now()
+		if lastTxnSendRound != curStatus.LastRound {
+			pongTx, err := pongClient.SendPaymentFromWallet(pongWalletHandle, nil, pongAccount, pingAccount, transactionFee, amountPongSendsPing, GenerateRandomBytes(8), "", 0, 0)
+			a.NoError(err, "fixture should be able to send money (pong -> ping)")
+			pongTxids = append(pongTxids, pongTx.ID().String())
 
-		pingTx, err := pingClient.SendPaymentFromUnencryptedWallet(pingAccount, pongAccount, transactionFee, amountPingSendsPong, GenerateRandomBytes(8))
-		a.NoError(err, "fixture should be able to send money (ping -> pong)")
-		pingTxids = append(pingTxids, pingTx.ID().String())
+			pingTx, err := pingClient.SendPaymentFromWallet(pingWalletHandle, nil, pingAccount, pongAccount, transactionFee, amountPingSendsPong, GenerateRandomBytes(8), "", 0, 0)
+			a.NoError(err, "fixture should be able to send money (ping -> pong)")
+			pingTxids = append(pingTxids, pingTx.ID().String())
 
-		expectedPingBalance = expectedPingBalance - transactionFee - amountPingSendsPong + amountPongSendsPing
-		expectedPongBalance = expectedPongBalance - transactionFee - amountPongSendsPing + amountPingSendsPong
+			expectedPingBalance = expectedPingBalance - transactionFee - amountPingSendsPong + amountPongSendsPing
+			expectedPongBalance = expectedPongBalance - transactionFee - amountPongSendsPing + amountPingSendsPong
+
+			lastTxnSendRound = curStatus.LastRound
+		}
 
 		curStatus, err = pongClient.Status()
 		a.NoError(err)
 
-		time.Sleep(time.Second)
+		pongWalletHandle, err = pongClient.GetUnencryptedWalletHandle()
```

Comment:
why this needs to be called at each iteration?

Reply:
The handle would expire after a while.. so it need to be refreshed frequently. I could have added a error-handling for that and refresh the handle only when needed, but I figured out that this would likely to be an overkill for this scenario.